Core/Src/main.c

```diff
@@ -75,6 +77,10 @@ int main(void)
 
   /* USER CODE BEGIN Init */
 
+  uint8_t T_buffer [3] = {0x01, 0x80, 0};
+  uint8_t R_buffer [3] = {0};
+  uint16_t adc_value = {0};
```

Comment:
`adc_value` should just be like this :

`uint16_t adc_value = 0;`

You don't need curly braces bec it is not an array

Reply:
IIRC, in ECE 150 they'll teach you to initialize values like this:
`uint16_t adc_value{0};`.

This is also valid, but you'll find that most people/companies prefer to use the "c-style" operation.